diff --git a/src/cts/src/HTreeBuilder.h b/src/cts/src/HTreeBuilder.h
index 8ee4aa3..45cf9ce 100644
--- a/src/cts/src/HTreeBuilder.h
+++ b/src/cts/src/HTreeBuilder.h
@@ -379,3 +379,7 @@ class HTreeBuilder : public TreeBuilder
 };
 
 }  // namespace cts
+#if defined(FMT_VERSION) && FMT_VERSION >= 90000
+#include <fmt/ostream.h>
+template <typename T> struct fmt::formatter<cts::Box<T>> : fmt::ostream_formatter {};
+#endif // FMT_VERSION >= 90000
diff --git a/src/cts/src/TreeBuilder.h b/src/cts/src/TreeBuilder.h
index bc3f63a..b298d33 100644
--- a/src/cts/src/TreeBuilder.h
+++ b/src/cts/src/TreeBuilder.h
@@ -265,3 +265,7 @@ class TreeBuilder
 };
 
 }  // namespace cts
+#if defined(FMT_VERSION) && FMT_VERSION >= 90000
+#include <fmt/ostream.h>
+template <typename T> struct fmt::formatter<cts::Point<T>> : fmt::ostream_formatter {};
+#endif // FMT_VERSION >= 90000
diff --git a/src/dpl/src/CheckPlacement.cpp b/src/dpl/src/CheckPlacement.cpp
index 5670949..191ba76 100644
--- a/src/dpl/src/CheckPlacement.cpp
+++ b/src/dpl/src/CheckPlacement.cpp
@@ -14,6 +14,7 @@
 #include "infrastructure/network.h"
 #include "odb/db.h"
 #include "utl/Logger.h"
+
 namespace dpl {
 
 using odb::Direction2D;
diff --git a/src/dpl/src/infrastructure/Grid.h b/src/dpl/src/infrastructure/Grid.h
index 2c239ed..5922edc 100644
--- a/src/dpl/src/infrastructure/Grid.h
+++ b/src/dpl/src/infrastructure/Grid.h
@@ -213,3 +213,11 @@ class Grid
 };
 
 }  // namespace dpl
+
+#if defined(FMT_VERSION) && FMT_VERSION >= 90000
+#include <fmt/ostream.h>
+template <> struct fmt::formatter<dpl::TypedCoordinate<dpl::GridYType>> : fmt::ostream_formatter {};
+template <> struct fmt::formatter<dpl::TypedCoordinate<dpl::DbuYType>> : fmt::ostream_formatter {};
+template <> struct fmt::formatter<dpl::TypedCoordinate<dpl::GridXType>> : fmt::ostream_formatter {};
+template <> struct fmt::formatter<dpl::TypedCoordinate<dpl::DbuXType>> : fmt::ostream_formatter {};
+#endif // FMT_VERSION >= 90000
diff --git a/src/drt/src/db/tech/frConstraint.h b/src/drt/src/db/tech/frConstraint.h
index 25dd827..8324111 100644
--- a/src/drt/src/db/tech/frConstraint.h
+++ b/src/drt/src/db/tech/frConstraint.h
@@ -2473,3 +2473,8 @@ class frNonDefaultRule
 };
 
 }  // namespace drt
+#if defined(FMT_VERSION) && FMT_VERSION >= 90000
+#include <fmt/ostream.h>
+template <> struct fmt::formatter<drt::frMinimumcutConnectionEnum> : fmt::ostream_formatter {};
+template <> struct fmt::formatter<drt::frCornerTypeEnum> : fmt::ostream_formatter {};
+#endif // FMT_VERSION >= 90000
diff --git a/src/drt/src/dr/FlexDR.h b/src/drt/src/dr/FlexDR.h
index 2d0687e..3d94d59 100644
--- a/src/drt/src/dr/FlexDR.h
+++ b/src/drt/src/dr/FlexDR.h
@@ -1110,3 +1110,8 @@ class FlexDRWorker
   friend class boost::serialization::access;
 };
 }  // namespace drt
+#if defined(FMT_VERSION) && FMT_VERSION >= 90000
+#include <fmt/ostream.h>
+template <> struct fmt::formatter<drt::frTime> : fmt::ostream_formatter {};
+template <> struct fmt::formatter<drt::frDirEnum> : fmt::ostream_formatter {};
+#endif // FMT_VERSION >= 90000
diff --git a/src/drt/src/dr/FlexDR_conn.h b/src/drt/src/dr/FlexDR_conn.h
index a100823..b8d75d6 100644
--- a/src/drt/src/dr/FlexDR_conn.h
+++ b/src/drt/src/dr/FlexDR_conn.h
@@ -184,3 +184,7 @@ class FlexDRConnectivityChecker
 };
 
 }  // namespace drt
+#if defined(FMT_VERSION) && FMT_VERSION >= 90000
+#include <fmt/ostream.h>
+template <> struct fmt::formatter<drt::frPathSeg> : fmt::ostream_formatter {};
+#endif // FMT_VERSION >= 90000
diff --git a/src/drt/src/frBaseTypes.h b/src/drt/src/frBaseTypes.h
index 2d01052..af377df 100644
--- a/src/drt/src/frBaseTypes.h
+++ b/src/drt/src/frBaseTypes.h
@@ -338,3 +338,7 @@ inline bool is_loading(const Archive& ar)
 using utl::format_as;
 
 }  // namespace drt
+#if defined(FMT_VERSION) && FMT_VERSION >= 90000
+#include <fmt/ostream.h>
+template <> struct fmt::formatter<drt::frConstraintTypeEnum> : fmt::ostream_formatter {};
+#endif // FMT_VERSION >= 90000
diff --git a/src/dst/src/BalancerConnection.cc b/src/dst/src/BalancerConnection.cc
index 602dabe..b515ff2 100644
--- a/src/dst/src/BalancerConnection.cc
+++ b/src/dst/src/BalancerConnection.cc
@@ -26,6 +26,11 @@
 #include "dst/Distributed.h"
 #include "utl/Logger.h"
 
+#if defined(FMT_VERSION) && FMT_VERSION >= 90000
+#include <fmt/ostream.h>
+template <> struct fmt::formatter<boost::asio::ip::address> : fmt::ostream_formatter {};
+#endif // FMT_VERSION >= 90000
+
 using namespace dst;
 
 BOOST_CLASS_EXPORT(dst::BalancerJobDescription)
diff --git a/src/dst/src/BalancerConnection.h b/src/dst/src/BalancerConnection.h
index 8352f9e..9bf8ade 100644
--- a/src/dst/src/BalancerConnection.h
+++ b/src/dst/src/BalancerConnection.h
@@ -47,3 +47,7 @@ class BalancerConnection
   static const int kMaxBroadcastFailedNodes = 2;
 };
 }  // namespace dst
+#if defined(FMT_VERSION) && FMT_VERSION >= 90000
+#include <fmt/ostream.h>
+template <> struct fmt::formatter<boost::asio::ip::address> : fmt::ostream_formatter {};
+#endif // FMT_VERSION >= 90000
diff --git a/src/dst/src/LoadBalancer.cc b/src/dst/src/LoadBalancer.cc
index 5250ff5..ae834c0 100644
--- a/src/dst/src/LoadBalancer.cc
+++ b/src/dst/src/LoadBalancer.cc
@@ -16,6 +16,10 @@
 #include "boost/bind/bind.hpp"
 #include "boost/thread/thread.hpp"
 #include "utl/Logger.h"
+#if defined(FMT_VERSION) && FMT_VERSION >= 90000
+#include <fmt/ostream.h>
+template <> struct fmt::formatter<boost::asio::ip::address> : fmt::ostream_formatter {};
+#endif // FMT_VERSION >= 90000
 
 using boost::asio::ip::udp;
 
diff --git a/src/dst/src/LoadBalancer.h b/src/dst/src/LoadBalancer.h
index a037ee9..5be8efd 100644
--- a/src/dst/src/LoadBalancer.h
+++ b/src/dst/src/LoadBalancer.h
@@ -89,3 +89,7 @@ class LoadBalancer
   friend class dst::BalancerConnection;
 };
 }  // namespace dst
+#if defined(FMT_VERSION) && FMT_VERSION >= 90000
+#include <fmt/ostream.h>
+template <> struct fmt::formatter<boost::asio::ip::address> : fmt::ostream_formatter {};
+#endif // FMT_VERSION >= 90000
diff --git a/src/grt/src/cugr/src/GRTree.h b/src/grt/src/cugr/src/GRTree.h
index ed6f196..c8304bd 100644
--- a/src/grt/src/cugr/src/GRTree.h
+++ b/src/grt/src/cugr/src/GRTree.h
@@ -52,3 +52,8 @@ class GRTreeNode : public GRPoint
 };
 
 }  // namespace grt
+
+#if defined(FMT_VERSION) && FMT_VERSION >= 90000
+#include <fmt/ostream.h>
+template <> struct fmt::formatter<grt::GRPoint> : fmt::ostream_formatter {};
+#endif // FMT_VERSION >= 90000
diff --git a/src/grt/src/cugr/src/GeoTypes.h b/src/grt/src/cugr/src/GeoTypes.h
index 7c99506..541cc25 100644
--- a/src/grt/src/cugr/src/GeoTypes.h
+++ b/src/grt/src/cugr/src/GeoTypes.h
@@ -36,3 +36,9 @@ class BoxOnLayer : public BoxT
 };
 
 }  // namespace grt
+
+
+#if defined(FMT_VERSION) && FMT_VERSION >= 90000
+#include <fmt/ostream.h>
+template <> struct fmt::formatter<grt::BoxOnLayer> : fmt::ostream_formatter {};
+#endif // FMT_VERSION >= 90000
diff --git a/src/grt/src/cugr/src/geo.h b/src/grt/src/cugr/src/geo.h
index 80e308f..f9afed7 100644
--- a/src/grt/src/cugr/src/geo.h
+++ b/src/grt/src/cugr/src/geo.h
@@ -358,3 +358,8 @@ class BoxT
 };
 
 }  // namespace grt
+
+#if defined(FMT_VERSION) && FMT_VERSION >= 90000
+#include <fmt/ostream.h>
+template <> struct fmt::formatter<grt::BoxT> : fmt::ostream_formatter {};
+#endif // FMT_VERSION >= 90000
diff --git a/src/grt/src/fastroute/include/DataType.h b/src/grt/src/fastroute/include/DataType.h
index 142f007..6039811 100644
--- a/src/grt/src/fastroute/include/DataType.h
+++ b/src/grt/src/fastroute/include/DataType.h
@@ -256,3 +256,8 @@ struct OrderNetEdge
 using utl::format_as;
 
 }  // namespace grt
+
+#if defined(FMT_VERSION) && FMT_VERSION >= 90000
+#include <fmt/ostream.h>
+template <> struct fmt::formatter<grt::RouteType> : fmt::ostream_formatter {};
+#endif // FMT_VERSION >= 90000
diff --git a/src/odb/include/odb/geom.h b/src/odb/include/odb/geom.h
index 2b682cd..19ffc9f 100644
--- a/src/odb/include/odb/geom.h
+++ b/src/odb/include/odb/geom.h
@@ -1091,3 +1091,8 @@ using utl::format_as;
 #endif
 
 }  // namespace odb
+#if defined(FMT_VERSION) && FMT_VERSION >= 90000
+#include <fmt/ostream.h>
+template <> struct fmt::formatter<odb::Point> : fmt::ostream_formatter {};
+template <> struct fmt::formatter<odb::Rect> : fmt::ostream_formatter {};
+#endif // FMT_VERSION >= 90000
diff --git a/src/utl/include/utl/timer.h b/src/utl/include/utl/timer.h
index 4a0390d..ed788da 100644
--- a/src/utl/include/utl/timer.h
+++ b/src/utl/include/utl/timer.h
@@ -55,3 +55,8 @@ class DebugScopedTimer : public Timer
 std::ostream& operator<<(std::ostream& os, const Timer& t);
 
 }  // namespace utl
+#if defined(FMT_VERSION) && FMT_VERSION >= 90000
+#include <fmt/ostream.h>
+template <> struct fmt::formatter<utl::DebugScopedTimer> : fmt::ostream_formatter {};
+template <> struct fmt::formatter<utl::Timer> : fmt::ostream_formatter {};
+#endif // FMT_VERSION >= 90000
