diff --git a/ihp-sg13g2/libs.tech/xschem/install.py b/ihp-sg13g2/libs.tech/xschem/install.py
index 1e8a188..580c0c4 100755
--- a/ihp-sg13g2/libs.tech/xschem/install.py
+++ b/ihp-sg13g2/libs.tech/xschem/install.py
@@ -82,9 +82,9 @@ if __name__ == "__main__":
     else:
         print(f"{program_name} is not installed.")
     	
-
-    original_file =  pdk_root + "/ihp-sg13g2/libs.tech/ngspice/.spiceinit"
-    symbolic_link = "/home/" + username + "/.spiceinit"
+    os.chdir(pdk_root + "/ihp-sg13g2/libs.tech/ngspice")
+    original_file = ".spiceinit"
+    symbolic_link = "spinit"
     # Create the symbolic link
     if not os.path.exists(symbolic_link):
         try:
