diff --git a/ihp-sg13g2/libs.tech/qucs/install.py b/ihp-sg13g2/libs.tech/qucs/install.py
index c903df0..f2c5afb 100755
--- a/ihp-sg13g2/libs.tech/qucs/install.py
+++ b/ihp-sg13g2/libs.tech/qucs/install.py
@@ -177,30 +177,6 @@ if __name__ == "__main__":
             logging.error(f"{program_name} is not installed.")
             exit(1)
                 
-    # Compiling Verilog-A models
-    print("\nCompiling Verilog-A models ...")
-    destination_directory = pdk_root + "/ihp-sg13g2/libs.tech/ngspice/osdi"
-    
-    if not os.path.exists(destination_directory):
-        os.makedirs(destination_directory)
-    
-    program_name = "openvaf"
-    if is_program_installed(program_name):
-        source_directory = pdk_root + "/ihp-sg13g2/libs.tech/verilog-a/psp103"
-        command = "openvaf psp103_nqs.va --output " + destination_directory + "/psp103_nqs.osdi"    
-        print(f"{program_name} is available and about to run the command '{command}' in a location: {source_directory} ")	
-        exec_app_in_directory(command, source_directory)
-        source_directory = pdk_root + "/ihp-sg13g2/libs.tech/verilog-a/r3_cmc"
-        command = "openvaf r3_cmc.va --output " + destination_directory + "/r3_cmc.osdi"    
-        print(f"{program_name} is available and about to run the command '{command}' in a location: {source_directory} ")	
-        exec_app_in_directory(command, source_directory)
-        source_directory = pdk_root + "/ihp-sg13g2/libs.tech/verilog-a/mosvar"
-        command = "openvaf mosvar.va --output " + destination_directory + "/mosvar.osdi"    
-        print(f"{program_name} is available and about to run the command '{command}' in a location: {source_directory} ")	
-        exec_app_in_directory(command, source_directory)
-    else:
-        logging.error(f"{program_name} is not installed.")
-        exit(1)
                 
                 
                 
diff --git a/ihp-sg13g2/libs.tech/xschem/install.py b/ihp-sg13g2/libs.tech/xschem/install.py
index 1e8a188..580c0c4 100755
--- a/ihp-sg13g2/libs.tech/xschem/install.py
+++ b/ihp-sg13g2/libs.tech/xschem/install.py
@@ -82,9 +82,9 @@ if __name__ == "__main__":
     else:
         print(f"{program_name} is not installed.")
     	
-
-    original_file =  pdk_root + "/ihp-sg13g2/libs.tech/ngspice/.spiceinit"
-    symbolic_link = "/home/" + username + "/.spiceinit"
+    os.chdir(pdk_root + "/ihp-sg13g2/libs.tech/ngspice")
+    original_file = ".spiceinit"
+    symbolic_link = "spinit"
     # Create the symbolic link
     if not os.path.exists(symbolic_link):
         try:
